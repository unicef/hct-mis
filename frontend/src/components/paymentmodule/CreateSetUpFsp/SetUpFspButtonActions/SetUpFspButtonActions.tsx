import { Box, Button } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

interface SetUpFspButtonActionsProps {
  step: number;
  submitForm: (values) => void;
  businessArea: string;
  paymentPlanId: string;
  handleBackStep: () => void;
}

export const SetUpFspButtonActions = ({
  step,
  submitForm,
  businessArea,
  paymentPlanId,
  handleBackStep,
}: SetUpFspButtonActionsProps): React.ReactElement => {
  const { t } = useTranslation();

  return (
    <Box pt={3} display='flex'>
      <Box mr={3}>
        {step === 0 && (
          <Button
            component={Link}
            to={`/${businessArea}/payment-module/payment-plan/${paymentPlanId}`}
          >
            {t('Cancel')}
          </Button>
        )}
        {step === 1 && <Button onClick={handleBackStep}>{t('Back')}</Button>}
      </Box>
      <Button variant='contained' color='primary' onClick={submitForm}>
        {t(step === 0 ? 'Next' : 'Save')}
      </Button>
    </Box>
  );
};
