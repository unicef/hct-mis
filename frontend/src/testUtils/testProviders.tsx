import MomentUtils from '@date-io/moment';
import { ThemeProvider } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import moment from 'moment';
import React from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>
    <StyledThemeProvider theme={theme}>
      <MuiPickersUtilsProvider libInstance={moment} utils={MomentUtils}>
        <CssBaseline />
        {children}
      </MuiPickersUtilsProvider>
    </StyledThemeProvider>
  </ThemeProvider>
);
