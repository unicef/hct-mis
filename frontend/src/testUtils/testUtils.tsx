import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
