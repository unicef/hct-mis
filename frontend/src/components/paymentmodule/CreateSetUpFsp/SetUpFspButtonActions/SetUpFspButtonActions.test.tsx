// import { act } from '@testing-library/react';
import React from 'react';
// import wait from 'waait';
import { render } from '../../../../testUtils/testUtils';
import { SetUpFspButtonActions } from './SetUpFspButtonActions';


describe('components/paymentmodule/CreateSetUpFsp/SetUpFspButtonActions', () => {
  it('should render', () => {
    const step = 0;
    const setStep = jest.fn();
    const submitForm = (_values) => Promise.resolve();
    const { container } = render(
      <SetUpFspButtonActions
        step={step}
        submitForm={submitForm}
        businessArea='afghanistan'
        paymentPlanId='asdjkfhsakdjfsd76asdf0sdf=='
        handleBackStep={setStep} />,
    );
    // await act(() => wait(0)); // wait for response
    expect(container).toMatchSnapshot();
  });
});
