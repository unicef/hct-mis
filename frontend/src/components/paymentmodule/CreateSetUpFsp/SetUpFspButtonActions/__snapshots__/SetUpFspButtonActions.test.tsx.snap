// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/CreateSetUpFsp/SetUpFspButtonActions should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/payment-module/payment-plans/asdjkfhsakdjfsd76asdf0sdf=="
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-cy="button-next-save"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
